avaframe/ana5Utils/distanceTimeAnalysis.py

```diff
@@ -0,0 +1,730 @@
+            radarFov[0][1], radarFov[1][0], radarFov[1][1])
+        log.error(message)
+        raise AssertionError(message)
+
+    # define radar field of view as dict
+    radarpath = {'x': np.asarray(radarFov[0]),
+                 'y': np.asarray(radarFov[1])}
+    # add z-coordinate from dem data
+    radarProfile = gT.projectOnRaster(demOriginal, radarpath, interp='bilinear')
+
+    # radar location - position of radar [0] and point in direction of field of view [1]
+    rX = radarProfile[0]['x'][0]
+    rY = radarProfile[0]['y'][0]
+    rZ = radarProfile[0]['z'][0]
+    # a point that gives direction - point in direction of field of view
+    rDx = radarProfile[0]['x'][1]
+    rDy = radarProfile[0]['y'][1]
+    rDz = radarProfile[0]['z'][1]
+
+    # convert centerline of radar field of view to spherical coordinates
+    rD, phiD, thetaD = gT.cartToSpherical(X=rDx-rX, Y=rDy-rY, Z=rDz-rZ)
+
+    # domain in spherical coordinates with center beeing radar position
+    r, phi, theta = gT.cartToSpherical(X=X-rX, Y=Y-rY, Z=rasterdata-rZ)
+
+    # create mask where in spherical coordinates azimuth angle is bigger than field of view of radar
+    # this is defined by the radar position where it is looking and the aperature angle
+    maskPhi = ~np.logical_and(phi > phiD - aperture, phi < phiD + aperture)
+
+    # compute radar range field (distance from radar -masked with field of view)
+    # TODO: check if we need to mask also where there are nans in dem
+    radarRange = np.ma.array(r, mask=maskPhi)
+
+    # create range gates
+    minR = int(np.floor(np.nanmin(radarRange)))
+    maxR = int(np.ceil(np.nanmax(radarRange)))
+    rgWidth = cfgRangeTime['GENERAL'].getfloat('rgWidth')
+    rangeGates= np.arange(minR+rgWidth/2, maxR+rgWidth/2, rgWidth)
+
+    return radarRange, rangeGates
+
+
+def minRangeSimulation(flowF, rangeMasked, threshold):
+    """ fetch info on the min range of simulation with respect to masked range array
+        the masked range array gives info what the visible range is
+
+        Parameters
+        -------------
+        flowF: numpy nd array
+            flow parameter result field
+        rangeMasked: masked numpy array
+            masked array of radar range
+        threshold: float
+            used to create mask for getting line of sight range,
+            only data above threshold taken into account
+
+        Returns
+        --------
+        losRange: float
+            minimum distance in line of sight to avalanche front
+            defined by the flowF
+
+    """
+
+    # fetch mask where result field not above threshold
+    maskAva = np.ma.masked_where(~(flowF > threshold), flowF)
+    bmaskAva = np.ma.getmask(maskAva)
+    # fetch mask of radar field of view
+    maskPhi = np.ma.getmask(rangeMasked)
+
+    # merge masks of result field not above threshold and outside radar field of view
+    maskFull = ~np.logical_and(~maskPhi, ~bmaskAva)
+
```

Comment:
we do the same here and in extractMeanValuesAtRangeGates. Could we try to do it only once?

Reply:
again here if we only use one of the two functions, it won't be available but I can put this in a function and call it from both

Reply:
but you always want to compute the distance to the front AND then extract the mean values no?

Reply:
so I made a function maskRangeFull that does these steps and is called from both, so that they can still be used individuall

Reply:
maybe not always both, in case you are just interested in the distance?

Reply:
Ok so now you put the mask stuff in a function, this is good but you still call the function twice... I would try to call the function outside and give the mask as input to minRangeSimulation and extractMeanValuesAtRangeGates.
My reasoning here is that if you say the ttdiagram thing is slow, we should avoid computing several times the same stuff.
Although I think what is slow is more the loop on the gates to get the mean value

Reply:
Ok now I shortened this drastically, if the fully masked range array is available it is actually only one line - so I included this in the extract mean values function - so calling the mask function only once now and removed two other functions